python_modules/libraries/dagster-postgres/dagster_postgres/resources.py

```diff
@@ -0,0 +1,122 @@
+    not close the connection on __exit__ - but rather only commits the current SQL transaction.
+    The dagster resource will automatically call connection.close() (but not connection.commit()),
+    after execution.
+
+    Examples:
+        .. code-block:: python
+
+            from dagster import Definitions, asset, EnvVar
+            from dagster_postgres import PostgresResource
+
+            @asset
+            def my_table(postgres: PostgresResource):
+                with postgres.get_connection() as conn:
+                    with conn.cursor() as cur:
+                        cur.execute("SELECT * FROM table;")
+
+            defs = Definitions(
+                assets=[my_table],
+                resources={
+                    "postgres": PostgresResource(
+                        host="localhost",
+                        port=5432,
+                        user="postgres",
+                        password=EnvVar("POSTGRES_PASSWORD")
+                    )
+                }
+            )
+
+
+    """
+
+    dsn: Optional[str] = Field(
+        description=(
+            "A libpq connection string."
+            " Can be provided together with keyword-arguments in a mix-and-match manner."
+            " Defaults to None if not provided - in which case only the keyword arguments"
+            " will be used to establish the connection."
+        ),
+        default=None,
+    )
+
+    host: Optional[str] = Field(
+        description=("Database host address. Defaults to UNIX socket if not provided"),
+        default=None,
+    )
+
+    port: Optional[int] = Field(
+        description=("Connection port number. defaults to 5432 if not provided."),
+        default=None,
+    )
+
+    user: Optional[str] = Field(
+        description=(
+            "User name used to authenticate."
+            " Defaults to None if not provided - in which case the 'user' field in"
+            " the dsn will be used"
+        ),
+        default=None,
+    )
+
+    password: Optional[str] = Field(
+        description=(
+            "Password used to authenticate."
+            " Defaults to None if not provided - in which case the 'password' field in"
+            " the dsn will be used"
+        ),
+        default=None,
+    )
+
+    database: Optional[str] = Field(
+        description=("The database name. Defaults to the user name if not provided"),
+        default=None,
+    )
+
+    additional_parameters: dict[str, Any] = Field(
+        description=(
+            "Additional parameters to pass to psycopg2.connect."
+            " For a full list of options, see"
+            " https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-PARAMKEYWORDS"
+        ),
+        default={},
+    )
+
+    _connection: psycopg2.extensions.connection = PrivateAttr()
+
+    def setup_for_execution(self, context: InitResourceContext) -> None:
+        self._connection = backoff(
```

Comment:
I don't think we want to initialize the connection here, but rather in the body of `get_connection`. When I call 
`with resource.get_connection() as conn`

I would expect that the connection is initialized in the call to `get_connection`. This is how most other resources work (see SnowflakeResource, SSHResource). I think it's confusing to initialize it and then just return it in the context manager, why use a context manager at all then?